### PR TITLE
Reorganize tests

### PR DESCRIPTION
**Pull request by sethaxen**
The current tests are a mess, and much of the code isn't tested. This PR completely reorganizes the tests, leaving a lot of empty test sets to mark where tests are needed. It doesn't add any tests; these should be handled in future PRs.

Because this PR changes every line of the test code, it also runs JuliaFormatter on the test files with BlueStyle.

### CODE REVIEW

File: test/Octonion.jl [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        end
+    end
+
+    @testset "basic" begin
+        real
+        imag
+        conj
+        float
+        Quaternions.abs_imag
+    end
```
Comment (hyrodium) [inline test/Octonion.jl:94]: Is this forgotten to add `@test`?

File: test/Octonion.jl [hunk 2]
```diff
@@ -0,0 +1,146 @@
+
```
Comment (hyrodium) [inline test/Octonion.jl:1]: Do we need this empty line?

File: test/Octonion.jl [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            test_group(q1, q2, q3, *, one(q), inv)
+            test_multiplicative(q1, q2, *, norm)
+
+            # complex embedding
+            test_multiplicative(c1, c2, *, Quaternion)
+            test_multiplicative(c1, c2, +, Quaternion)
+        end
+    end
+
+    @testset "isreal" begin end
```
Comment (hyrodium) [inline test/Octonion.jl:117]: Is this still WIP?

**Review by hyrodium — COMMENTED** [review]
Is this still WIP?

File: test/Octonion.jl (hunk 1 above)
Comment (sethaxen) [reply test/Octonion.jl]: I left these here to note which functions should be tested here, but I've replaced this with a #TODO comment

File: test/Octonion.jl (hunk 3 above)
Comment (sethaxen) [reply test/Octonion.jl:112]: No, as noted in the OP, I left empty test sets to mark where tests should be added. The goal this PR is a pure restructure, not to add new tests.

Comment (hyrodium) [reply test/Octonion.jl:112]: Ah, sorry that I overlooked the description.

File: test/Octonion.jl [hunk 4]
```diff
@@ -0,0 +1,141 @@
+using Quaternions
+using LinearAlgebra
+using Random
+using Test
+
+@testset "Quaternion" begin
```
Comment (hyrodium) [inline test/Octonion.jl:6]:
```suggestion
@testset "Octonion" begin
```

File: test/Quaternion.jl [hunk 5]
```diff
@@ -0,0 +1,442 @@
+using Quaternions
+using LinearAlgebra
+using Random
+using Test
```
Comment (hyrodium) [inline test/Quaternion.jl:4]: Do we need these `using` here? I thought this can be moved to `runtests.jl`.

Comment (hyrodium) [reply test/Quaternion.jl:4]: Ah, I was always commenting out the `include` in `runtests.jl` for that. :smile:

**Review by hyrodium — APPROVED** [review]
>85.32% <0.00%> (-0.55%)

I'm not sure the reason for the change of the coverage rate, but we can add more tests in another PR.

EDIT: https://app.codecov.io/gh/JuliaGeometry/Quaternions.jl/compare/72/changes#D1L15

**Comment by sethaxen** [comment]
> > 85.32% <0.00%> (-0.55%)
> 
> I'm not sure the reason for the change of the coverage rate, but we can add more tests in another PR.
> 
> EDIT: https://app.codecov.io/gh/JuliaGeometry/Quaternions.jl/compare/72/changes#D1L15

Yes, this is because `sample` was using this constructor, but it was never explicitly tested. All the constructors need tests in a later PR.

**Pull request merged**